precompiles/assets-erc20/example/README.md

```diff
@@ -0,0 +1,92 @@
+    constructor(IERC20 burrito)
+    Xc20Plus("Wrapped Burrito", "xcBUR")
+    ERC20Permit("Wrapped Burrito")
+    ERC20Wrapper(burrito)
+    {}
+
+    function _mint(address _to, uint256 _amount)
+        internal
+        override(ERC20)
+    {
+        // add here your pre-mint hooks if needed
+
+        require(
+            IERC20Plus(xcBurrito).mint(_to, _amount), "Minting xc token failed"
+        );
+
+        // add here your post-mint hooks if needed
+    }
+}
+```
+Check the full example on the Github repo.
+## Procedures
+### 1. Create XC20 asset
+Follow the documentation on how to [Create XC20 Assets](https://docs.astar.network/docs/xcm/building-with-xcm/create-xc20-assets)
+
+### 2. Deploy XcBurrito.sol smart contract
+To deploy XcBurrito contract you need 2 input parameters
+- Burrito ERC20 token address (H160)
+- XC20 asset address (H160)
+    - Follow instructions on how to [Generate XC20 Address](https://docs.astar.network/docs/xcm/integration/tools). If your asset_id is 17 (=0x11), your EVM address will be `0xffffffff00000000000000000000000000000011`
+
+### 3. Transfer xc asset ownership to the XcBurrito smart contract
+To enable xcBurrito contract to be able to mint/burn you need to call `setTeam()` extrinsic on pallet-assets. Since the EVM contract has 24 bytes (H160), we need to transform EVM address into SS58 format to be able to call this extrinsic.
+<img width="1205" alt="Screenshot 2022-08-04 at 09 32 11" src="https://user-images.githubusercontent.com/34627453/182795050-f10dad6e-dbd9-4ac9-b526-d51a66503cb9.png">
+After you set the team you should renounce the ownership over the xc20 asset by calling `transferOwnership()` extrinsic on pallet-assets.
+
+<img width="1205" alt="Screenshot 2022-08-04 at 09 32 41" src="https://user-images.githubusercontent.com/34627453/182795097-d145073b-cdf0-4c28-916f-66abb5c68abc.png">
+
+Now XcBurrito contract can start minting xc20 tokens for the callers who own Burrito ERC20 token.
+
+## User actions
+To convert Burrito token into XcBurrito assets, users will need to do two actions.
+1. User Approves XcBurrito smart contract to transfer ERC20 Burrito tokens from user's balance to the XcBurrito contract. As an alternative use `permit()` call to improve UX
+2. User calls XcBurrito `depositFor()` for minting new xc20 tokens or `withdrawTo()` for burning xc20 tokens
+
+## End result
+The result for `depositFor(user, amount)` will be:
+- XcBurrito locks `amount` of ERC20 Burrito token
+- User's balance for ERC20 Buritto is decreased for `amount`
+- XcBurrito contract mints `amount` of XcBurrito asset
+- User's balance for XcBurrito asset is increased for `amount`
+
+
+The result for `withdrawTo(user, amount)` will be:
+- XcBurrito contract burns `amount` of XcBurrito asset
+- User's balance for XcBurrito asset is decreased for `amount`
+- XcBurrito contract sends `amount` of ERC20 Burrito token to user
+- User's balance for ERC20 Burrito is increased for `amount`
+
+
+
```

Comment:
For the finish, we should state that now wrapped token can be sent to other parachains.